File: sharedComposables/src/commonMain/kotlin/dev/mateusz1913/f1results/composable/common/DriverStandingRow.kt
package dev.mateusz1913.f1results.composable.common

import androidx.compose.foundation.layout.*
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import dev.mateusz1913.f1results.datasource.data.standings.DriverStandingType

@Composable
fun DriverStandingRow(driverStanding: DriverStandingType, noDriverInfo: Boolean = false) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Row(
            modifier = Modifier.weight(1f),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            StandingPosition(driverStanding.positionText)
            Column(modifier = Modifier.weight(1f)) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    if (!noDriverInfo) {
                        Text(
                            "${driverStanding.driver.givenName} ${driverStanding.driver.familyName}",
                            modifier = Modifier.padding(horizontal = 10.dp, vertical = 2.dp),
                            fontSize = 18.sp,
                            fontWeight = FontWeight.Bold,
                        )
                    }
                }
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    val constructorsNames = driverStanding.constructors.joinToString(", ") { it.name }
                    Text(
                        constructorsNames,
                        fontSize = 18.sp,
                        fontWeight = FontWeight.SemiBold,
                        modifier = Modifier.padding(
                            start = if (!noDriverInfo) 10.dp else 0.dp,
                            top = 2.dp,
                            end = 10.dp,
                            bottom = 2.dp
                        )
                    )
                }
            }
        }
        StandingScore(driverStanding.points, driverStanding.wins)
    }
}

File: swiftApp/Shared/Presentation/Common/DriverStandingRow.swift
import SwiftUI
import shared

struct DriverStandingRow: View {
    var driverStanding: DriverStandingType
    var noDriverInfo: Bool = false
    
    var body: some View {
        HStack(alignment: .center) {
            HStack(alignment: .center) {
                StandingPosition(position: driverStanding.positionText)
                VStack(alignment: .center) {
                    HStack(alignment: .center) {
                        if !noDriverInfo {
                            Text("\(driverStanding.driver.givenName) \(driverStanding.driver.familyName)")
                                .font(.system(size: 18))
                                .fontWeight(.bold)
                                .padding(.init(top: 2, leading: 10, bottom: 2, trailing: 10))
                                .frame(minWidth: 0, maxWidth: .infinity, alignment: .leading)
                        }
                    }
                    .fillMaxWidth()
                    HStack(alignment: .center) {
                        let constructorsNames = getConstructorsNames(driverStanding.constructors)
                        Text(constructorsNames)
                            .font(.system(size: 18))
                            .fontWeight(.semibold)
                            .padding(.init(top: 2, leading: !noDriverInfo ? 10 : 0, bottom: 2, trailing: 10))
                            .frame(minWidth: 0, maxWidth: .infinity, alignment: .leading)
                    }
                    .fillMaxWidth()
                }
                .fillMaxSize()
            }
            .fillMaxSize()
            StandingScore(points: driverStanding.points, wins: driverStanding.wins)
        }
        .fillMaxWidth()
    }
    
    private func getConstructorsNames(_ constructors: KotlinArray<ConstructorType>) -> String {
        var constructorsNamesArr = [String]()
        for index in 0..<constructors.size {
            if let constructor = constructors.get(index: index) {
                constructorsNamesArr.append(constructor.name)
            }
        }
        return constructorsNamesArr.joined(separator: " / ")
    }
}
